--- old/qtbase/configure	2016-05-25 22:46:17.000000000 +0700
+++ new/qtbase/configure	2020-03-31 17:08:52.952549481 +0700
@@ -543,7 +543,7 @@
         exit 2
     fi
 
-    if ! /usr/bin/xcrun -find xcrun >/dev/null 2>&1; then
+    if ! /usr/bin/xcrun -find xcodebuild >/dev/null 2>&1; then
         echo >&2
         echo "   Xcode not set up properly. You may need to confirm the license" >&2
         echo "   agreement by running /usr/bin/xcodebuild without arguments." >&2
--- old/qtbase/mkspecs/features/mac/default_pre.prf	2016-05-25 22:46:17.000000000 +0700
+++ new/qtbase/mkspecs/features/mac/default_pre.prf	2020-03-24 20:54:27.256768958 +0700
@@ -12,7 +12,7 @@
         error("Xcode is not installed in $${QMAKE_XCODE_DEVELOPER_PATH}. Please use xcode-select to choose Xcode installation path.")
 
     # Make sure Xcode is set up properly
-    isEmpty($$list($$system("/usr/bin/xcrun -find xcrun 2>/dev/null"))): \
+    isEmpty($$list($$system("/usr/bin/xcrun -find xcodebuild 2>/dev/null"))): \
         error("Xcode not set up properly. You may need to confirm the license agreement by running /usr/bin/xcodebuild.")
 }
 
--- old/qtbase/src/platformsupport/fontdatabases/mac/qfontengine_coretext.mm	2016-05-25 22:46:17.000000000 +0700
+++ new/qtbase/src/platformsupport/fontdatabases/mac/qfontengine_coretext.mm	2020-03-30 19:55:22.322886577 +0700
@@ -769,7 +769,7 @@
 
 QFixed QCoreTextFontEngine::emSquareSize() const
 {
-    return QFixed::QFixed(int(CTFontGetUnitsPerEm(ctfont)));
+    return QFixed(int(CTFontGetUnitsPerEm(ctfont)));
 }
 
 QFontEngine *QCoreTextFontEngine::cloneWithSize(qreal pixelSize) const
--- old/qtbase/src/plugins/platforms/cocoa/qcocoahelpers.mm	2016-05-25 22:46:17.000000000 +0700
+++ new/qtbase/src/plugins/platforms/cocoa/qcocoahelpers.mm	2020-03-30 20:42:16.122865534 +0700
@@ -542,12 +542,6 @@
 OSStatus qt_mac_drawCGImage(CGContextRef inContext, const CGRect *inBounds, CGImageRef inImage)
 {
     // Verbatim copy if HIViewDrawCGImage (as shown on Carbon-Dev)
-    OSStatus err = noErr;
-
-    require_action(inContext != NULL, InvalidContext, err = paramErr);
-    require_action(inBounds != NULL, InvalidBounds, err = paramErr);
-    require_action(inImage != NULL, InvalidImage, err = paramErr);
-
     CGContextSaveGState( inContext );
     CGContextTranslateCTM (inContext, 0, inBounds->origin.y + CGRectGetMaxY(*inBounds));
     CGContextScaleCTM(inContext, 1, -1);
@@ -555,10 +549,8 @@
     CGContextDrawImage(inContext, *inBounds, inImage);
 
     CGContextRestoreGState(inContext);
-InvalidImage:
-InvalidBounds:
-InvalidContext:
-        return err;
+
+    return noErr;
 }
 
 Qt::MouseButton cocoaButton2QtButton(NSInteger buttonNum)
